code-challenges/challenges-05.test.js

```diff
@@ -0,0 +1,366 @@
+'use strict';
+
+// to learn more about the cheerio library and what it is doing, look at their documentation: https://www.npmjs.com/package/cheerio
+const cheerio = require('cheerio');
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named templateWithJQuery that uses jQuery to get the html template from the DOM, copy the contents, fill it with the Star Wars People, and append it to the DOM.
+------------------------------------------------------------------------------------------------ */
+let starWarsPeople = [
+  {
+    "name": "Luke Skywalker",
+    "height": "172",
+    "eye_color": "blue"
+  },
+  {
+    "name": "C-3PO",
+    "height": "167",
+    "eye_color": "yellow"
+  },
+  {
+    "name": "R2-D2",
+    "height": "96",
+    "eye_color": "red"
+  }
+];
+
+let $ = createSnippetWithJQuery(`
+<main>
+  <section id="template">
+    <h2></h2>
+    <h3></h3>
+    <p></p>
+  </section>
+</main>
+`);
+
+const templateWithJQuery = () => {
+  // Solution code here...
+  let $template = $('#template').html();
+
+  starWarsPeople.forEach((item) => {
+
+    let $templateClone = $('<section></section>');
+    $templateClone.html($template);
+
+
+    $templateClone.find('h2').text(item.name);
+    $templateClone.find('h3').text(item.height);
+    $templateClone.find('p').text(item.eye_color);
+
+    $('main').append($templateClone);
+  });
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named howMuchPencil that takes in a string, as written on the side of a pencil.
+
+As you sharpen the pencil, the string will become shorter and shorter, starting by removing the first letter.
+
+Your function should use slice within a loop and return an array of each successive string result from losing letters to the sharpener, until nothing is left.
+
+For example, if the input is 'Welcome', the output will be:
+['Welcome', 'elcome', 'lcome', 'come', 'ome', 'me', 'e', ''].
+------------------------------------------------------------------------------------------------ */
+
+const howMuchPencil = (str) => {
+  let result = [];
+  // Solution code here...
+  result.push(str);
```

Comment:
This line here is what I think is causing you issues. You don't want to push anything into the array, until after you iterate through it in your loop.